File: frontend/domain/repositories/IDraftRepository.ts
import {
  CreateDraftPayloadEntity,
  UpdateChaptersDraftPayloadEntity,
  DraftEntity,
  UpdateLessonsDraftPayloadEntity,
} from "../entities/draft.entity";

export interface IDraftRepository {
  createDraft(
    payload: CreateDraftPayloadEntity,
    accessToken: string,
    refreshToken: string,
  ): Promise<string>;
  getDraft(
    draftId: string,
    accessToken: string,
    refreshToken: string,
  ): Promise<DraftEntity>;
  updateChapters(
    payload: UpdateChaptersDraftPayloadEntity,
    accessToken: string,
    refreshToken: string,
  ): Promise<boolean>;
  updateLessons(
    payload: UpdateLessonsDraftPayloadEntity,
    accessToken: string,
    refreshToken: string,
  ): Promise<boolean>;
  generateMatrix(
    draftId: string,
    accessToken: string,
    refreshToken: string,
  ): Promise<boolean>;
  generateMatrixDetails(
    draftId: string,
    accessToken: string,
    refreshToken: string,
  ): Promise<boolean>;
  getRecentDraft(
    accessToken: string,
    refreshToken: string,
  ): Promise<DraftEntity[]>;
  getAllUserDrafts(
    accessToken: string,
    refreshToken: string,
  ): Promise<DraftEntity[]>;
  deleteDraft(
    draftId: string,
    accessToken: string,
    refreshToken: string,
  ): Promise<boolean>;
}

File: backend/src/main/java/com/wedu/exam_creation/draft/domain/repository/IDraftRepository.java
package com.wedu.exam_creation.draft.domain.repository;

import com.wedu.exam_creation.draft.domain.entity.DraftEntity;
import com.wedu.exam_creation.draft.domain.payload.UpdateChaptersPayload;
import com.wedu.exam_creation.draft.domain.payload.UpdateLessonsPayload;
import com.wedu.exam_creation.draft.domain.payload.UpdateMatrixDetailsPayload;
import com.wedu.exam_creation.draft.domain.payload.UpdateMatrixPayload;

import java.util.List;

public interface IDraftRepository {
    String createDraft(DraftEntity draft);

    DraftEntity getDraft(String draftId, String userId);

    boolean updateChapters(UpdateChaptersPayload payload);

    boolean updateLessons(UpdateLessonsPayload payload);

    boolean updateMatrix(List<UpdateMatrixPayload> payloads);

    boolean updateMatrixDetails(List<UpdateMatrixDetailsPayload> payloads);

    boolean deleteDraft(String draftId, String userId);

    List<DraftEntity> getRecentDraft(String userId);

    List<DraftEntity> getAllUserDrafts(String userId);
}
